client/src/components/AppointmentDataForm.js

```diff
@@ -0,0 +1,94 @@
+// @flow
+
+import "react-datetime/css/react-datetime.css";
+
+import { gql, useMutation, useQuery } from "@apollo/client";
+
+import AppointmentForm from "./AppointmentForm";
+import FETCH_APPOINTMENTS from "./fetchAppointments";
+import React from "react";
+import { useHistory } from "react-router-dom";
+
+const CREATE_APPOINTMENT = gql`
+  mutation createAppointment(
+    $startTime: Int!
+    $endTime: Int!
+    $address: String!
+    $note: String!
+  ) {
+    createAppointment(
+      startTime: $startTime
+      endTime: $endTime
+      address: $address
+      note: $note
+    ) {
+      id
+    }
+  }
+`;
+
+const FETCH_APPOINTMENT = gql`
+  query Appointment($id: ID!) {
+    appointment(id: $id) {
+      id
+      startTime
+      endTime
+      location {
+        id
+        address
+      }
+    }
+  }
+`;
+
+const AppointmentDataForm: () => React$Node = (props) => {
+  const history = useHistory();
+  const backToLanding = () => {
+    history.push("/");
+  };
+
+  if (props.match === undefined) {
+    const Create = () => {
+      const [
+        createAppointment,
+        { loading: mutationLoading, error: mutationError },
+      ] = useMutation(CREATE_APPOINTMENT, {
+        refetchQueries: [{ query: FETCH_APPOINTMENTS }],
+        onCompleted: () => backToLanding(),
+      });
+
+      return (
+        <div>
+          <AppointmentForm appointmnetUpdate={createAppointment} />
```

Comment:
appointmnetUpdate is a typo

Reply:
Oh, sorry. I did not notice that. Thanks.